### PR TITLE
ref(project_upstream): Revert fallback to v2 requests

### PR DESCRIPTION
**Pull request by flub**
The v3 endpoint was not very reliable and we used to fallback to 2v,
now the endpoint is fixed we no longer need this.  Without the
fallback the behaviour of relay is much easier to understand.

This reverts #1314

#skip-changelog

### CODE REVIEW

**Review by untitaker — APPROVED** [review]
💯 👯‍♀️ 🚢 🇮🇹 ♋

**Review by jjbayer — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 - Fall back to version 2 project config if version 3 fails. ([#1314](https://github.com/getsentry/relay/pull/1314))
 - Reduce number of metrics extracted for release health. ([#1316](https://github.com/getsentry/relay/pull/1316))
 - Indicate with thread is the main thread in thread metadata for profiles. ([#1320](https://github.com/getsentry/relay/pull/1320))
+- Revert fallback to v2 project config, v3 is reliable now. ([#1325](https://github.com/getsentry/relay/pull/1325))
```
Comment (jjbayer) [inline CHANGELOG.md]: nit: Not sure what our policy is here, but when we revert a change within the same unreleased version, I would just remove both changelog items.

Comment (untitaker) [reply CHANGELOG.md]: i would agree with that actually

**Review by iker-barriocanal — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (flub) [reply CHANGELOG.md]: I don't know, I removed them now.

Personally I wouldn't even keep internal changelog entries at all, so I just followed what was there.

**Pull request merged**